das/store.go

```diff
@@ -0,0 +1,114 @@
+package das
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	"golang.org/x/net/context"
+)
+
+var (
+	storePrefix   = datastore.NewKey("das")
+	checkpointKey = datastore.NewKey("checkpoint")
+)
+
+type checkpointStore interface {
+	load(ctx context.Context) (checkpoint, error)
+	store(ctx context.Context, cp checkpoint)
```

Comment:
why no error returned?

Reply:
IMO this should still return an error and leave it up to caller to decide what to do with it.